lib/std/mem.zig

```diff
@@ -10,16 +10,67 @@ const testing = std.testing;
 const Endian = std.builtin.Endian;
 const native_endian = builtin.cpu.arch.endian();
 
+pub const PageSizeCheck = enum {
+    /// Does nothing if the page size is invalid
+    unsafe,
+    /// Throws an error if the page size is invalid
+    throw,
+    /// Asserts if the page size is invalid
+    assert,
+    /// Panics if the page size is invalid
+    panic,
+
+    pub fn makeType(comptime self: PageSizeCheck) type {
+        return switch (self) {
+            .assert, .unsafe, .panic => usize,
+            .throw => error{InvalidPageSize}!usize,
+        };
+    }
+};
+
+pub const PageSizeOptions = struct {
+    /// Value sanity check method
+    check: PageSizeCheck = .assert,
+    /// Use the comptime version of page size if the OS does not report it.
+    comptime_fallback: bool = true,
+};
+
+/// Runtime known minimum page size.
+pub fn getPageSize(comptime options: PageSizeOptions) PageSizeCheck.makeType(options.check) {
+    const value: usize = switch (builtin.os.tag) {
+        .linux => if (builtin.link_libc) @intCast(std.c.sysconf(std.os.linux.SC.PAGESIZE)) else std.os.linux.getauxval(std.elf.AT_PAGESZ),
```

Comment:
std.mem should not make any syscalls. The new logic should go into std.heap instead.

Reply:
Wouldn't it make more sense to go in `std.os`? Will rename.

Reply:
Keep in mind that `std.os` will be renamed to `std.posix` (#5019). Page size is not a concept limited to posix.

Reply:
Alright, it's in heap.